src/bp/core/messaging/messaging-webhooks.ts

```diff
@@ -0,0 +1,130 @@
+import axios from 'axios'
+import { IO } from 'botpress/sdk'
+import { EventEngine } from 'core/events'
+import { TYPES } from 'core/types'
+import { inject, injectable, postConstruct } from 'inversify'
+import { AppLifecycle, AppLifecycleEvents } from 'lifecycle'
+import { ConversationService } from './conversation-service'
+import { MessageService } from './message-service'
+
+@injectable()
+export class MessagingWebhooks {
+  constructor(
+    @inject(TYPES.EventEngine) private eventEngine: EventEngine,
+    @inject(TYPES.ConversationService) private conversationService: ConversationService,
+    @inject(TYPES.MessageService) private messageService: MessageService
+  ) {}
+
+  @postConstruct()
+  async init() {
+    await AppLifecycle.waitFor(AppLifecycleEvents.CONFIGURATION_LOADED)
+
+    // TODO: register webhooks by http
+    const incomingWebhooks = [
+      {
+        id: 'mfkt837834her',
+        url: 'http://localhost:3020/hook',
+        // TODO: This could be a regex (but regex is kind of garbadge to write)
+        // This scope could be even further scoped for a bot: 'app.myBotId.userMessage'
+        scopes: ['app.userMessage']
```

Comment:
Scopes will defined which type of webhook "event" triggers and http request. Filtering preferably happens before we make the request since it costs a lot more to post to a wehbook that does nothing than to call a middleware that gets skipped.